From a17ffd00b7813f5e18539fdd1202878d2582da87 Mon Sep 17 00:00:00 2001
From: Matt Johnston <matt@codeconstruct.com.au>
Date: Tue, 22 Feb 2022 12:29:36 +0800
Subject: [PATCH backport 5.10 59/73] mctp: Fix warnings reported by
 clang-analyzer

net/mctp/device.c:140:11: warning: Assigned value is garbage or undefined
[clang-analyzer-core.uninitialized.Assign]
        mcb->idx = idx;

- Not a real problem due to how the callback runs, fix the warning.

net/mctp/route.c:458:4: warning: Value stored to 'msk' is never read
[clang-analyzer-deadcode.DeadStores]
        msk = container_of(key->sk, struct mctp_sock, sk);

- 'msk' dead assignment can be removed here.

Signed-off-by: Matt Johnston <matt@codeconstruct.com.au>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 8d783197f06d905e5e7a89342e815ef5aeaa1731)
---
 net/mctp/device.c | 2 +-
 net/mctp/route.c  | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/mctp/device.c b/net/mctp/device.c
index 21b5ec022b16..787d00c95599 100644
--- a/net/mctp/device.c
+++ b/net/mctp/device.c
@@ -119,7 +119,7 @@ static int mctp_dump_addrinfo(struct sk_buff *skb, struct netlink_callback *cb)
 	struct ifaddrmsg *hdr;
 	struct mctp_dev *mdev;
 	int ifindex;
-	int idx, rc;
+	int idx = 0, rc;
 
 	hdr = nlmsg_data(cb->nlh);
 	// filter by ifindex if requested
diff --git a/net/mctp/route.c b/net/mctp/route.c
index 867de72b36df..a5efcae6a719 100644
--- a/net/mctp/route.c
+++ b/net/mctp/route.c
@@ -455,7 +455,6 @@ static int mctp_route_input(struct mctp_route *route, struct sk_buff *skb)
 		 * the reassembly/response key
 		 */
 		if (!rc && flags & MCTP_HDR_FLAG_EOM) {
-			msk = container_of(key->sk, struct mctp_sock, sk);
 			sock_queue_rcv_skb(key->sk, key->reasm_head);
 			key->reasm_head = NULL;
 			__mctp_key_done_in(key, net, f, MCTP_TRACE_KEY_REPLIED);
-- 
2.20.1

